sql/analyzer/generate_index_scans.go

```diff
@@ -161,30 +161,42 @@ func pushdownIndexesToTable(scope *Scope, a *Analyzer, tableNode sql.NameableNod
 				table = tw.Underlying()
 			}
 			if _, ok := table.(sql.IndexAddressableTable); ok {
-				indexLookup, ok := indexes[tableNode.Name()]
-				if ok && indexLookup.lookup.Index.CanSupport(indexLookup.lookup.Ranges...) {
-					a.Log("table %q transformed with pushdown of index", tableNode.Name())
-					ita, err := plan.NewStaticIndexedAccessForResolvedTable(n, indexLookup.lookup)
-					if plan.ErrInvalidLookupForIndexedTable.Is(err) {
-						return n, transform.SameTree, nil
-					}
-					if err != nil {
-						return nil, transform.SameTree, err
-					}
+				if indexLookup, ok := indexes[tableNode.Name()]; ok {
+					if indexLookup.lookup.Index.IsFullText() {
+						ftFilterTable, err := createFulltextFilterTable(ctx, indexLookup.expr, n)
```

Comment:
We are resolving full text indexes twice in two differently places, and both of them are at the end of analysis. The way this needs to work with new name resolution is that MatchAgainst is resolved first, and in pushdown we already have a valid MatchAgainst. A maybe compromise that doesn't put the burden of rewriting this whole thing on one of us, depending on who merges first, would be to get a `resolveMatchAgainst` into onceBefore. The same rule would be inlined into `planbuilder/scalar.go` when we build the original MatchAgainst expression.